### PR TITLE
[MRG+1] Add template for a downloader middleware

### PR DESCRIPTION
**Pull request by redapple**
While testing downloader middleware with media pipeline (for https://github.com/scrapy/scrapy/issues/2753#issuecomment-303678711), I looked for a sample downloader middleware and found only a spider middleware template.

I personally never remember what methods to implement, nor what they should return, hence the verbose comments in the template.

Related/alternative https://github.com/scrapy/scrapy/issues/2633

### CODE REVIEW

File: scrapy/templates/project/module/middlewares.py.tmpl [hunk 1]
```diff
@@ -54,3 +54,50 @@ class ${ProjectName}SpiderMiddleware(object):
...
+        # Must either:
+        # - return None: continue processing this request
+        # - or return a Response object
+        # - or return a Request object
+        # - or raise IgnoreRequest: process_exception() methods of
+        #   installed downloader middleware will be called
+        return None
+
+    def process_response(self, request, response, spider):
+        # Called with the respsonse returned from the downloader.
```
Comment (aperezalbela) [inline scrapy/templates/project/module/middlewares.py.tmpl]: s/respsonse/response

File: scrapy/templates/project/module/middlewares.py.tmpl [hunk 2]
```diff
@@ -54,3 +54,50 @@ class ${ProjectName}SpiderMiddleware(object):
...
+
+        # Must either;
+        # - return a Response object
+        # - return a Request object
+        # - or raise IgnoreRequest
+        return response
+
+    def process_exception(self, request, exception, spider):
+        # Called when a download handler or a process_request()
+        # (from other downloader middelware) raises and exception.
```
Comment (aperezalbela) [inline scrapy/templates/project/module/middlewares.py.tmpl]: s/middelware/middleware

**Comment by redapple** [comment]
Thanks for spotting those typos @aperezalbela !

**Comment by dangra** [comment]
LGTM.

**Pull request merged**